pkg/storage/stores/local/shipper.go

```diff
@@ -164,15 +169,21 @@ func (s *Shipper) loop() {
 	for {
 		select {
 		case <-resyncTicker.C:
+			status := statusSuccess
 			err := s.syncLocalWithStorage(context.Background())
 			if err != nil {
+				status = statusFailure
 				level.Error(pkg_util.Logger).Log("msg", "error syncing local boltdb files with storage", "err", err)
 			}
+			s.metrics.filesDownloadOperationTotal.WithLabelValues(status).Inc()
```

Comment:
a defer could do a better job but that's fine !

Reply:
I agree with @cyriltovena here -- defer would help with some code deduplication.